kiss

```diff
@@ -1543,19 +1543,7 @@ pkg_update() {
 
     # Update each repository in '$KISS_PATH'.
     for repo do
-        if git -C "$repo" remote >/dev/null 2>&1; then
-            repo_type=git
-
-            # Get the Git repository root directory.
-            subm=$(git -C "$repo" rev-parse --show-superproject-working-tree)
-            repo=$(git -C "${subm:-"$repo"}" rev-parse --show-toplevel)
```

Comment:
This moves the git filtering code to each hook. ie, hooks get the list without duplicates (../repo/core -> ../repo). Without the filtering hooks will run git commands multiple times for a single git repository. Maybe we provide generic git, fossil, etc hooks which work for 99% of cases?

Reply:
It would be nice to provide a `post-updates` hook that gets executed once at the end of all updates, as well as a `updates-fail` or `update-fail`, so we can cleanup stuff if need be. See https://github.com/jedahan/kiss-repo-repos/pull/1/files?diff=unified&w=1#diff-e127914727aa9940998fabc10874225d27c633210c1d707c9391a7a81376390a for an example use-case

Reply:
As far as I know, `fossil` doesn't really have a built-in way like `git` does to only get the root of the working tree; `fossil info` will output it, but you'd need to do some parsing (`fossil info | grep local-root | cut -d ':' -f 2` would print it (with some leading spaces)).